src/serializer/ResidualHeapSerializer.js

```diff
@@ -1692,13 +1692,25 @@ export class ResidualHeapSerializer {
     let remainingProperties = new Map(val.properties);
     const dummyProperties = new Set();
     let props = [];
+    let isCertainlyLeaked = !val.mightNotBeHavocedObject();
+
+    // TODO #2259: Make deduplication in the face of leaking work for custom accessors
+    let shouldDropAsAssignedProp = (descriptor: Descriptor) =>
+      isCertainlyLeaked && !IsDataDescriptor(descriptor);
```

Comment:
The original logic for this seemed more consistent to "isCertainlyLeaked && IsDataDescriptor(descriptor)".